pretyping/tacred.ml

```diff
@@ -1009,11 +1009,11 @@ let change_map_constr_with_binders_left_to_right g f (env, l as acc) sigma c =
     let app = (mkApp (hdf, Array.sub al 0 (Array.length al - 1))) in
     let app' = f acc app in
     let a' = f acc a in
-      (match EConstr.kind sigma app' with
-      | App (hdf', al') when hdf' == hdf ->
-        (* Still the same projection, we ignore the change in parameters *)
-        mkProj (p, a')
-      | _ -> mkApp (app', [| a' |]))
+    let hdf', _ = decompose_app_vect sigma app' in
+    if hdf' == hdf then
```

Comment:
I know you did not introduce this, but the use of physical equality looks very fragile to me. Can't you use a more robust heuristic?

Reply:
In particular, `hdf` is guaranteed to be a `Const` node, you could take advantage of that.

Reply:
I don't really want to mess with it.